bokehjs/src/coffee/embed.ts

```diff
@@ -88,8 +102,27 @@ function _init_comms(target: string, doc: Document): void {
     } catch (e) {
       logger.warn(`Jupyter comms failed to register. push_notebook() will not function. (exception reported: ${e})`)
     }
-  } else
-    console.warn('Jupyter notebooks comms not available. push_notebook() will not function');
+  } else if (doc.roots()[0].id in kernels) {
+    logger.info(`Registering JupyterLab comms for target ${target}`)
+    const kernel = kernels[doc.roots()[0].id]
+    for (const id in kernel._comms) {
+      const promise = kernel._comms[id]
+      promise.then(update_comms)
+    }
+    try {
+      kernel.registerCommTarget(target, (comm: Comm) => {
+        logger.info(`Registering JupyterLab comms for target ${target}`)
+        const r = new Receiver()
+        comm.onMsg = _handle_notebook_comms.bind(doc, r)
+      })
+    } catch (e) {
+      logger.warn(`Jupyter comms failed to register. push_notebook() will not function. (exception reported: ${e})`)
+    }
+  } else {
+    console.warn('Jupyter notebooks comms not available. push_notebook() will not function.'
+                 'If running JupyterLab ensure the latest jupyterlab_bokeh extension is installed.'
+				 'In an exported notebook this warning is expected.');
```

Comment:
This isn't valid syntax. Also, indentation is off.

Reply:
Thanks, I did wonder, but how come this compiles and runs okay?

Reply:
It doesn't compile:
```
[18:21:20] src/coffee/embed.ts(123,18): error TS1005: ',' expected.
[18:21:20] src/coffee/embed.ts(124,6): error TS1005: ',' expected.
```
Our build currently doesn't fail if there are compilation errors. This will change with PR #7560.

Reply:
It also doesn't run correctly, at least according to chrome, and fails with:
```
SyntaxError: missing ) after argument list
```

Reply:
Great, good to know, thanks! Easy to overlook errors locally, so I'm looking forward to having it error.

Reply:
You can use string template literals (backticks instead of quotes) for multiline strings, but that doesn't work nicely with indented text, so usually is pretty useless without additional processing.

Reply:
>SyntaxError: missing ) after argument list

Suspect it must have cached an old version when I tried it.